sympy/assumptions/facts.py

```diff
@@ -34,71 +36,138 @@ def get_composite_predicates():
+
+    """
     # We build the facts starting with primitive predicates.
     # DO NOT include the predicates in get_composite_predicates()'s keys here!
-    return And(
-
+    x = Symbol('x')
+    fact = And(
         # primitive predicates exclude each other
-        Implies(Q.negative_infinite, ~Q.positive_infinite),
-        Implies(Q.negative, ~Q.zero & ~Q.positive),
-        Implies(Q.positive, ~Q.zero),
+        Implies(Q.negative_infinite(x), ~Q.positive_infinite(x)),
+        Implies(Q.negative(x), ~Q.zero(x) & ~Q.positive(x)),
+        Implies(Q.positive(x), ~Q.zero(x)),
 
         # build real line and complex plane
-        Implies(Q.negative | Q.zero | Q.positive, ~Q.imaginary),
-        Implies(Q.negative | Q.zero | Q.positive | Q.imaginary, Q.algebraic | Q.transcendental),
+        Implies(Q.negative(x) | Q.zero(x) | Q.positive(x), ~Q.imaginary(x)),
+        Implies(Q.negative(x) | Q.zero(x) | Q.positive(x) | Q.imaginary(x),
+                Q.algebraic(x) | Q.transcendental(x)),
 
         # other subsets of complex
-        Implies(Q.transcendental, ~Q.algebraic),
-        Implies(Q.irrational, ~Q.rational),
-        Equivalent(Q.rational | Q.irrational, Q.negative | Q.zero | Q.positive),
-        Implies(Q.rational, Q.algebraic),
+        Implies(Q.transcendental(x), ~Q.algebraic(x)),
+        Implies(Q.irrational(x), ~Q.rational(x)),
+        Equivalent(Q.rational(x) | Q.irrational(x),
+                Q.negative(x) | Q.zero(x) | Q.positive(x)),
+        Implies(Q.rational(x), Q.algebraic(x)),
 
         # integers
-        Implies(Q.even, ~Q.odd),
-        Implies(Q.even | Q.odd, Q.rational),
-        Implies(Q.zero, Q.even),
-        Implies(Q.composite, ~Q.prime),
-        Implies(Q.composite | Q.prime, (Q.even | Q.odd) & Q.positive),
-        Implies(Q.even & Q.positive & ~Q.prime, Q.composite),
+        Implies(Q.even(x), ~Q.odd(x)),
+        Implies(Q.even(x) | Q.odd(x), Q.rational(x)),
+        Implies(Q.zero(x), Q.even(x)),
+        Implies(Q.composite(x), ~Q.prime(x)),
+        Implies(Q.composite(x) | Q.prime(x),
+            (Q.even(x) | Q.odd(x)) & Q.positive(x)),
+        Implies(Q.even(x) & Q.positive(x) & ~Q.prime(x), Q.composite(x)),
 
         # hermitian and antihermitian
-        Implies(Q.negative | Q.zero | Q.positive, Q.hermitian),
-        Implies(Q.imaginary, Q.antihermitian),
-        Implies(Q.zero, Q.hermitian | Q.antihermitian),
+        Implies(Q.negative(x) | Q.zero(x) | Q.positive(x),
+            Q.hermitian(x)),
+        Implies(Q.imaginary(x), Q.antihermitian(x)),
+        Implies(Q.zero(x), Q.hermitian(x) | Q.antihermitian(x)),
 
         # define finity and infinity, and build extended real line
-        Implies(Q.infinite, ~Q.finite),
-        Implies(Q.algebraic | Q.transcendental, Q.finite),
-        Implies(Q.negative_infinite | Q.positive_infinite, Q.infinite),
+        Implies(Q.infinite(x), ~Q.finite(x)),
+        Implies(Q.algebraic(x) | Q.transcendental(x), Q.finite(x)),
```

Comment:
This is not needed if we recognised that `Q.complex` is equivalent and implies finite.